src/js/firebase-imp.ts

```diff
@@ -53,11 +55,13 @@ class FirebaseImp {
   connectionStatus: FirebaseRef
   dataRef: FirebaseRef
   userRef: FirebaseRef
+  sessionExists:boolean
```

Comment:
It looks like we never update `sessionExists` with `true` -- maybe we don't need to keep it as an instance var?  I am fine with removing it or updating it.